### PR TITLE
Use client go

### PR DESCRIPTION
**Pull request by awh**
Ease #16 by obviating the need to vendor kubernetes entire.

### CODE REVIEW

File: cmd/weave-npc/main.go
Comment (bboreham) [inline cmd/weave-npc/main.go]: `v1` seems a rather meaning-free name; maybe alias to something like `k8sv1` ?

Comment (awh) [reply cmd/weave-npc/main.go]: ¯_(ツ)_/¯ blame the api machinery folks for that :) `apiv1` makes more sense IMO

Comment (awh) [reply cmd/weave-npc/main.go]: This also begs the question of what should be done with `apis/extensions`...

**Comment by awh** [comment]
@bboreham refactored to use `apiv1` and `apiext` package aliases.

**Comment by awh** [comment]
I am getting this:

```
panic: interface conversion: interface {} is *v1beta1.NetworkPolicy, not *extensions.NetworkPolicy
```

If I change things to use `v1beta1`, it no longer compiles due to the fact that we have versioned and unversioned LabelSelectors:

```
../go/src/github.com/weaveworks/weave-npc/pkg/controller/analyser.go:22: cannot use &policy.Spec.PodSelector (type *v1beta1.LabelSelector) as type *unversioned.LabelSelector in argument to newSelectorSpec
../go/src/github.com/weaveworks/weave-npc/pkg/controller/analyser.go:59: cannot use peer.PodSelector (type *v1beta1.LabelSelector) as type *unversioned.LabelSelector in argument to newSelectorSpec
../go/src/github.com/weaveworks/weave-npc/pkg/controller/analyser.go:66: cannot use peer.NamespaceSelector (type *v1beta1.LabelSelector) as type *unversioned.LabelSelector in argument to newSelectorSpec
```

**Comment by awh** [comment]
This is waiting for kubernetes/kubernetes#35255 to land in `client-go`.

**Comment by awh** [comment]
Reworked against the current `client-go` `master` (`b22087a53becae45931ed72d5e0f12e0031d771a` as of writing) which contains the requisite `unversioned.LabelSelector` change. Once this PR is merged, I will start a new PR against `weaveworks/weave` to integrate the code there with vendoring.

PTAL @bboreham

**Review by bboreham — APPROVED** [review]
LGTM

**Pull request merged**
